src/core/permissions/reducers/getDataSetPermissionsPageReducer.js

```diff
@@ -0,0 +1,37 @@
+/*
+ * @flow
+ */
+
+import { Map } from 'immutable';
+import { RequestStates } from 'redux-reqseq';
+import type { SequenceAction } from 'redux-reqseq';
+
+import { DATA_SET_PERMISSIONS_PAGE, ERROR, REQUEST_STATE } from '../../redux/constants';
+import { GET_DATA_SET_PERMISSIONS_PAGE, getDataSetPermissionsPage } from '../actions';
+
+export default function reducer(state :Map, action :SequenceAction) {
+
+  return getDataSetPermissionsPage.reducer(state, action, {
+    REQUEST: () => state
+      .setIn([GET_DATA_SET_PERMISSIONS_PAGE, REQUEST_STATE], RequestStates.PENDING)
+      .setIn([GET_DATA_SET_PERMISSIONS_PAGE, action.id], action),
+    SUCCESS: () => {
+      if (state.hasIn([GET_DATA_SET_PERMISSIONS_PAGE, action.id])) {
+        return state
+          .set(DATA_SET_PERMISSIONS_PAGE, action.value)
+          .setIn([GET_DATA_SET_PERMISSIONS_PAGE, REQUEST_STATE], RequestStates.SUCCESS);
+      }
+      return state;
+    },
+    FAILURE: () => {
+      if (state.hasIn([GET_DATA_SET_PERMISSIONS_PAGE, action.id])) {
+        return state
+          .set(DATA_SET_PERMISSIONS_PAGE, Map())
+          .setIn([GET_DATA_SET_PERMISSIONS_PAGE, ERROR], action.value)
+          .setIn([GET_DATA_SET_PERMISSIONS_PAGE, REQUEST_STATE], RequestStates.FAILURE);
+      }
+      return state;
```

Comment:
Do you need an initial state for this reducer? Looks like `GET_DATA_SET_PERMISSIONS_PAGE` should initially have a STANDBY requestState.

Reply:
It does have an initial state, set in `src/core/permissions/reducers/index.js`

```
[GET_DATA_SET_PERMISSIONS_PAGE]: RS_INITIAL_STATE,
```